ENIGMAsystem/SHELL/Universal_System/roomsystem.cpp

```diff
@@ -537,89 +526,64 @@ int room_set_view_enabled(int indx, int val)
 
 }
 
-namespace enigma
-{
-  bool tile_alter = false;
-  bool instance_alter = false;
-}
-
 namespace enigma_user
 {
 
-int room_tile_add_ext(int indx, int bck, int left, int top, int width, int height, int x, int y, int depth, int xscale, int yscale, double alpha, int color)
+int room_tile_add_ext(int indx, int bck, int left, int top, int width, int height, int x, int y, int depth, double xscale, double yscale, double alpha, int color)
 {
   errcheck(indx,"Nonexistent room", 0);
   enigma::roomstruct *rm = enigma::roomdata[indx];
-  const int tcount = rm->tilecount++;
-  enigma::tile *ti = rm->tiles;
-  enigma::tile *newtiles = new enigma::tile[tcount + 1];
-  for (int tilei = 0; tilei < tcount; tilei++)
-    newtiles[tilei] = ti[tilei];
-
-  newtiles[tcount].id = enigma::maxtileid++;
-  newtiles[tcount].bckid = bck;
-  newtiles[tcount].bgx = left;
-  newtiles[tcount].bgy = top;
-  newtiles[tcount].depth = depth;
-  newtiles[tcount].height = height;
-  newtiles[tcount].width = width;
-  newtiles[tcount].roomX = x;
-  newtiles[tcount].roomY = y;
-  newtiles[tcount].xscale = xscale;
-  newtiles[tcount].yscale = yscale;
-  newtiles[tcount].alpha = alpha;
-  newtiles[tcount].color = color;
-
-  if (enigma::tile_alter) delete[] rm->tiles;
+
+  std::vector<enigma::tile> newtiles = rm->tiles;
+
+  newtiles.emplace_back(
+    enigma::maxtileid++,
+    bck,
+    left,
+    top,
+    depth,
+    height,
+    width,
+    x,
+    y,
+    xscale,
+    yscale,
+    alpha,
+    color
+  );
+
   rm->tiles = newtiles;
```

Comment:
This line and the `newtiles` line can both be deleted and just emplace into `rm->tiles`. Either that or delete just this line and make `newtiles` a const reference.

Reply:
Ok: 2a6feea5e8cda61b4d00b061121adf82fe0f7594

I removed it myself and tested.